internal/engine/compiler/impl_amd64_test.go

```diff
@@ -441,6 +441,61 @@ func TestAmd64Compiler_preventCrossedTargetdRegisters(t *testing.T) {
 	}
 }
 
+// Relates to #1111: older AMD64 CPUs do not support the LZCNT instruction
+// CPUID should be used instead. We simulate presence/absence of the feature
+// by overriding the field in the corresponding struct.
+func TestAmd64Compiler_ensureClz_ABM(t *testing.T) {
+	tests := []struct {
+		name        string
+		cpuFeatures cpuFeatureFlags
+		expectedLen int
+	}{
+		{
+			name:        "with ABM",
+			expectedLen: 140,
```

Comment:
rather than length, let's check the actual hex just like we've done in asm/amd64 package tests